packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -1360,12 +1608,12 @@ private IfStatement CreateConditionalSerializationStatement(
         }
 
         /// <summary>
-        /// Builds the JSON write core body statement for the additional raw data.
+        /// Builds the JSON write core body statement for the additional binary data.
         /// </summary>
-        /// <returns>The method body statement that writes the additional raw data.</returns>
+        /// <returns>The method body statement that writes the additional binary data.</returns>
         private MethodBodyStatement CreateWriteAdditionalRawDataStatement()
         {
-            if (_rawDataField == null)
+            if (_rawDataField == null || _additionalBinaryDataProperty != null)
```

Comment:
Why do we need both checks? Can we have the property without the backing field?

Reply:
The backing field will always be set if the model supports binary data AP. My reasoning for this was to handle the case where a derived model doesn't support AP. In that case, the base model would have the rawdata field serialization block and we wouldn't want to duplicate it. In contrary, if a model does support AP of binary data, then we don't generate this serialization block as the serialization will be handled as part of serializing the additional properties.

Reply:
Do we have a test case for this?

Reply:
Discussed offline. The cadle ranch test suite has several models that exercise this scenario. Resolving.